docs/pages/setup/reference/predicate-language.mdx

```diff
@@ -0,0 +1,67 @@
+---
+title: Predicate Language
+description: How to use predicate language used to define filter conditions.
+h1: Predicate Language
+---
+
+Predicate language is used to define filter or where condition for select resources.
+It is also used as a query language to filter and search through a [list of select resources](#resource-filtering).
+
+The predicate language support slightly differs depending on where it is used as explained in the next sections.
+
+## Defining Filter or Where Conditions in Role Resources
+
+Some fields in our role resources uses the predicate language:
+
+- [Dynamic Impersonation](./access-controls/guides/impersonation/#dynamic-impersonation)
+- [RBAC for sessions](./access-controls/reference/#rbac-for-sessions)
```

Comment:
the predicate language is also used for `access requests`, but i could not find documentation on how to define the fields (so i left it out for now). there are mentions of the field here though but with no examples: https://goteleport.com/docs/setup/reference/terraform-provider/#teleport_role, 

![image](https://user-images.githubusercontent.com/43280172/158225941-90699517-5f46-458e-a64f-e4d577ed51f5.png)

i'm not sure if these fields are only applicable to terraform provider?